frontend_app_admin_workspace_user/src/component/AddUserForm.jsx

```diff
@@ -28,55 +43,114 @@ export class AddUserForm extends React.Component {
-      : [state.newUserName, state.newUserEmail, state.newUserPassword, state.newUserProfile].some(i => i === '')
+      ? (
+        state.newUserName === '' ||
+        ((state.newUserUsername === '' || !props.newUsernameAvailability) && state.newUserEmail === '') ||
+        state.newUserProfile === ''
+      )
+      : (
+        state.newUserName === '' ||
+        ((state.newUserUsername === '' || !props.newUsernameAvailability) && state.newUserEmail === '') ||
+        state.newUserProfile === '' ||
+        state.newUserPassword === ''
+      )
   }
 
   render () {
     const { props, state } = this
 
     return (
       <form className='adminUser__adduser__form' data-cy='adminUser__adduser__form'>
-        <div className='adminUser__adduser__form__username'>
-          <label className='username__text' htmlFor='adduser_name'>
+        <div className='adminUser__adduser__form__userData'>
+          <label className='userData__text' htmlFor='adduser_name'>
             {props.t('Full name')}
           </label>
 
           <input
             type='text'
-            className='username__input form-control'
+            className='userData__input form-control'
             id='adduser_name'
             placeholder={props.t('Full name')}
             value={state.newUserName}
             onChange={this.handleChangeNewUserName}
             data-cy='adduser_name'
           />
 
-          <label className='username__text' htmlFor='adduser_email'>
-            {props.t('Email')}
+          <label className='userData__text' htmlFor='adduser_username'>
+            {props.t('Username')}
           </label>
 
           <input
             type='text'
-            className='username__input form-control'
+            className='userData__input form-control'
+            id='adduser_username'
+            placeholder={props.t('@username')}
+            value={state.newUserUsername}
+            onChange={this.handleChangeNewUserUsername}
+            data-cy='adduser_username'
+          />
+
+          {!props.newUsernameAvailability && state.newUserUsername !== '' && (
+            <div className='userData__input__username__errorMsg'>
+              <i className='userData__input__username__errorIcon fa fa-times' />
+              {props.t('This username is not available')}
+            </div>
+          )}
+
+          <div className='userData__email'>
+            <label className='userData__text' htmlFor='adduser_email'>
+              {props.t('Email')}
+            </label>
+
+            <button
+              type='button'
+              className='userData__email__info'
+              id='popoverEmailInfo'
+            >
+              <i className='fa fa-fw fa-question-circle' />
+            </button>
+
+            <Popover
+              placement='bottom'
+              isOpen={state.popoverEmailInfoOpen}
+              target='popoverEmailInfo'
+              toggle={this.handleTogglePopoverEmailInfo}
```

Comment:
with new linting rules, you need to add `// eslint-disable-line react/jsx-handler-names` otherwise it'll tell you to rename `toggle` into `onToggle` but you can't sine it's from a lirary

Reply:
Changed